torch/csrc/utils/schema_info.cpp

```diff
@@ -262,24 +263,38 @@ std::vector<c10::FunctionSchema> SchemaInfo::getNonDeterministicOps() {
-  };
+std::vector<SchemaSpecialCasePair> SchemaInfo::getTrainingOps() {
+  // This is a list of pairs of ops to sets of strings
+  //  where the a boolean variable (either "training",
+  // "train" or "use_input_stats") affects the mutability
+  // of the unorderered set of strings.
+  static const std::vector<std::pair<std::string, std::unordered_set<std::string>>> training_op_pairs =
+      {{"aten::batch_norm(Tensor input, Tensor? weight, Tensor? bias, Tensor? running_mean, Tensor? running_var, bool training, float momentum, float eps, bool cudnn_enabled) -> Tensor",
+        {"running_mean", "running_var"}},
+       {"aten::instance_norm(Tensor input, Tensor? weight, Tensor? bias, Tensor? running_mean, Tensor? running_var, bool use_input_stats, float momentum, float eps, bool cudnn_enabled) -> Tensor",
+        {"running_mean", "running_var"}},
+       {"aten::_batch_norm_impl_index(Tensor input, Tensor? weight, Tensor? bias, Tensor? running_mean, Tensor? running_var, bool training, float momentum, float eps, bool cudnn_enabled) -> (Tensor, Tensor, Tensor, Tensor, int)",
+        {"running_mean", "running_var"}},
+       {"aten::cudnn_batch_norm(Tensor input, Tensor weight, Tensor? bias, Tensor? running_mean, Tensor? running_var, bool training, float exponential_average_factor, float epsilon) -> (Tensor, Tensor, Tensor, Tensor)",
+        {"running_mean", "running_var"}},
+       {"aten::miopen_batch_norm(Tensor input, Tensor weight, Tensor? bias, Tensor? running_mean, Tensor? running_var, bool training, float exponential_average_factor, float epsilon) -> (Tensor, Tensor, Tensor)",
+        {"running_mean", "running_var"}},
+       {"aten::native_batch_norm(Tensor input, Tensor? weight, Tensor? bias, Tensor? running_mean, Tensor? running_var, bool training, float momentum, float eps) -> (Tensor, Tensor, Tensor)",
+        {"running_mean", "running_var"}},
+       {"aten::native_batch_norm.out(Tensor input, Tensor? weight, Tensor? bias, Tensor? running_mean, Tensor? running_var, bool training, float momentum, float eps, *, Tensor(a!) out, Tensor(b!) save_mean, Tensor(c!) save_invstd) -> (Tensor(a!), Tensor(b!), Tensor(c!))",
+        {"running_mean", "running_var"}},
+       {"aten::rrelu_with_noise(Tensor self, Tensor noise, Scalar lower=0.125, Scalar upper=0.3333333333333333, bool training=False, Generator? generator=None) -> Tensor",
+        {"noise"}},
+       {"aten::rrelu_with_noise.out(Tensor self, Tensor(b!) noise, Scalar lower=0.125, Scalar upper=0.3333333333333333, bool training=False, Generator? generator=None, *, Tensor(a!) out) -> Tensor(a!)",
+        {"noise"}},
+       {"rrelu_with_noise_(Tensor(a!) self, Tensor(b!) noise, Scalar lower=0.125, Scalar upper=0.3333333333333333, bool training=False, Generator? generator=None) -> Tensor(a!)",
```

Comment:
should these not have (b!) annotating the noise param? I think this change hasn't been added to native_functions.yaml yet

Reply:
Oh I must've copy pasted the wrong version of the schemas. I'll change this.